### ISSUE TITLE
Pylance(reportUnknownVariableType) when enumerate() is used with parameter start

### ISSUE THREAD

**Issue by hflamme** [microsoft/pylance-release#2127]
## Environment data

-   Language Server version: 2021.11.2
-   OS and version: Ubuntu 20.04.3 LTS
-   Python 3.9.5

## Expected behaviour

Within the following code
```python
a: list[int] = [1, 2, 3, 4]
b = [c for i, c in enumerate(a, start=1)]
```
the type of `b` should be inferred as `list[int]`.
This is the case when removing the start argument, i.e. for
```python
a: list[int] = [1, 2, 3, 4]
b = [c for i, c in enumerate(a)]
print(b)
```
types are asserted just fine,

## Actual behaviour

`Pylance(reportUnknownVariableType)` for variable `b` and `c` (see below).

## Logs

![grafik](https://user-images.githubusercontent.com/44997732/144192190-1f6209a6-46e8-4388-8fed-0c0aad2ffefc.png)

**Comment by erictraut** [comment]
Thanks for reporting the problem. I think this has been fixed since the last release of pylance. I'm able to repro it with the last released version, but I'm no longer able to repro it with the latest code. A new version of pylance will be released within the next day or two, and that should address the problem.

**Comment by hflamme** [comment]
Perfect, thanks

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.12.0, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021120-2-december-2021)
